astropy/config/paths.py

```diff
@@ -105,25 +41,25 @@ def get_config_dir(rootname="astropy"):
     # If using set_temp_config, that overrides all
     if set_temp_config._temp_path is not None:
         xch = set_temp_config._temp_path
-        config_path = os.path.join(xch, rootname)
-        if not os.path.exists(config_path):
-            os.mkdir(config_path)
-        return os.path.abspath(config_path)
+        config_path = xch / rootname
+        config_path.mkdir(exist_ok=True)
+        return str(config_path.resolve())
 
     # first look for XDG_CONFIG_HOME
-    xch = os.environ.get("XDG_CONFIG_HOME")
+    if (
+        (xdg_config_home := os.getenv("XDG_CONFIG_HOME")) is not None
+        and (xch := Path(xdg_config_home)).exists()
+        and not (xchpth := xch / rootname).is_symlink()
+    ):
+        if xchpth.exists():
+            return str(xchpth.resolve())
+        else:
+            linkto = xchpth
 
-    if xch is not None and os.path.exists(xch):
-        xchpth = os.path.join(xch, rootname)
-        if not os.path.islink(xchpth):
-            if os.path.exists(xchpth):
-                return os.path.abspath(xchpth)
-            else:
-                linkto = xchpth
-    return os.path.abspath(_find_or_create_root_dir("config", linkto, rootname))
+    return str(_find_or_create_root_dir("config", linkto, rootname))
 
 
-def get_cache_dir(rootname="astropy"):
+def get_cache_dir(rootname: str = "astropy") -> str:
```

Comment:
Is this function essentially the same as `get_config_dir` ?
Maybe a good pre or followup PR is to make a private function which these both call. Consolidate the code.

Reply:
Good call. I'll make sure to do the follow up too !